ui/console/writer.go

```diff
@@ -0,0 +1,41 @@
+package console
+
+import (
+	"bytes"
+	"io"
+	"os"
+	"sync"
+)
+
+// Writer syncs writes with a mutex and, if the output is a TTY, clears before
+// newlines.
+type Writer struct {
```

Comment:
This is the same `consoleWriter`, but since it needed to be exported so that it could be used in `cmd/tests/` and elsewhere, it's the only thing in this new `ui/console` package. Otherwise the functionality is identical, and we can bring `console.Console` back later.